test_runner/regress/test_compute_migrations.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, cast
+
+import pytest
+from fixtures.compute_migrations import NUM_COMPUTE_MIGRATIONS
+
+if TYPE_CHECKING:
+    from fixtures.neon_fixtures import NeonEnv
+
+
+MIGRATIONS: tuple[str, ...] = (
+    "0001-neon_superuser_bypass_rls",
+    "0002-alter_roles",
+    "0003-grant_pg_create_subscription_to_neon_superuser",
+    "0004-grant_pg_monitor_to_neon_superuser",
+    "0005-grant_all_on_tables_to_neon_superuser",
+    "0006-grant_all_on_sequences_to_neon_superuser",
+    "0007-grant_all_on_tables_to_neon_superuser_with_grant_option",
+    "0008-grant_all_on_sequences_to_neon_superuser_with_grant_option",
+    "0009-revoke_replication_for_previously_allowed_roles",
+    "0010-grant_snapshot_synchronization_funcs_to_neon_superuser",
+    "0011-grant_pg_show_replication_origin_status_to_neon_superuser",
+)
+
+
+def test_migrations_smoke(neon_simple_env: NeonEnv, compute_migrations_dir: Path):
+    """
+    Test that migrations ran and succeeded, or were skipped.
+    """
+    env = neon_simple_env
+
+    endpoint = env.endpoints.create("main")
+    endpoint.respec(skip_pg_catalog_updates=False)
+    # Tell compute_ctl to error when the second migration is about to run, so
+    # that can test the restart functionality. We would typically want to do
+    # this via the HTTP endpoint, but we can't guarantee that we will be able to
+    # send our HTTP request before the migrations are ran.
+    endpoint.start(env={"FAILPOINTS": "the-second-compute-migration=return"})
+
+    # Make sure that the first migration ran
+    endpoint.wait_for_migrations(wait_for=1)
```

Comment:
Suggestion: Follow up to the comment about parametrizing the failpoint. Maybe it's an overkill, but I'm thinking about making this a loop over all migrations:

```python
for i in NUM_COMPUTE_MIGRATIONS:
  1. Start with failpoint till i
  2. Check that we applied till i-1

3. Start without failpoints and check that we applied till the end
```

If later we will have multi-DB migrations, like we wanted for CVE fix, this will help to verify that migrations are retriable and idempotent

What do you think?

(if it's hard to parametrize the failpoint, then it's fine to postpone, this is already a step forward)

Reply:
In my next push, let me know what you think.